### PR TITLE
minor fixes & tools improvements

### PR DESCRIPTION
**Pull request by WaberZhuang**
**What this PR does / why we need it**:
- [bugfix] libcurl->set_user_passwd(...) may cause basic and bearer auth coexist when the registry's authentication domain name is the same as the registry's domain name (e.g. hub.docker.com/auth/...)
- disable huge page
- cmake 3.15 -> 3.14
- [feat] supports SSL/TLS client certs
- [feat] overlaybd-apply: streaming download and apply
- [feat] overlaybd-commit: streaming zfile compress and upload. This is unavailable with option `-t` because it only supports write-in-sequence, but we have to overwrite the tar header.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Please check the following list**:
- [ ]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [x]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

File: CMake/Findphoton.cmake [hunk 1]
```diff
@@ -5,7 +5,7 @@ set(PHOTON_ENABLE_EXTFS ON)
 FetchContent_Declare(
   photon
   GIT_REPOSITORY https://github.com/alibaba/PhotonLibOS.git
-  GIT_TAG v0.6.14
+  GIT_TAG 1df8b3e9fa11cb3f495841ad88d9bfb74c180f07
```
Comment (liulanzheng) [inline CMake/Findphoton.cmake:8]: v0.6.15 is ready to be used

File: src/overlaybd/registryfs/registryfs.cpp [hunk 2]
```diff
@@ -134,9 +138,9 @@ class RegistryFSImpl : public RegistryFS {
 
         {
             auto curl = get_cURL();
-            if (photon::net::http::what_protocol(actual_url) == 2) {
-                curl->reset().clear_header().setopt(CURLOPT_SSL_VERIFYPEER, 0L);
-            }
+            // if (photon::net::http::what_protocol(actual_url) == 2) {
```
Comment (liulanzheng) [inline src/overlaybd/registryfs/registryfs.cpp:141]: remove?

File: src/overlaybd/registryfs/registryfs.cpp [hunk 3]
```diff
@@ -334,7 +347,10 @@ class RegistryFSImpl : public RegistryFS {
         DEFER({ release_cURL(req); });
         photon::net::StringWriter writer;
         if (!username.empty()) {
-            req->set_user_passwd(username.c_str(), password.c_str()).set_redirect(3);
+            // req->set_user_passwd(username.c_str(), password.c_str()).set_redirect(3);
```
Comment (liulanzheng) [inline src/overlaybd/registryfs/registryfs.cpp:350]: What are the differences from the previous version? does set_user_passwd also use basic auth？

File: src/overlaybd/registryfs/registryfs_v2.cpp [hunk 4]
```diff
@@ -110,11 +113,45 @@ class RegistryFSImpl_v2 : public RegistryFS {
...
+            char buf[4096];
+            ret = file->pread(buf, st.st_size, 0);
+            if (ret != st.st_size) {
+                LOG_ERRNO_RETURN(0, -1, "failed to read file `", fn);
+            }
+            target = estring(buf);
+            return 0;
+        };
+
+        readFile(cert_file, m_cert_str);
```
Comment (liulanzheng) [inline src/overlaybd/registryfs/registryfs_v2.cpp:150]: need to check the result?

File: src/overlaybd/registryfs/registryfs.cpp (hunk 3 above)
Comment (WaberZhuang) [reply src/overlaybd/registryfs/registryfs.cpp:350]: set_user_passwd will always set basic auth and libcurl->reset() does not work.

Comment (liulanzheng) [inline src/overlaybd/registryfs/registryfs.cpp:350]: remove this line

File: src/overlaybd/registryfs/registryfs_v2.cpp [hunk 5]
```diff
@@ -768,14 +787,65 @@ class RegistryUploader : public VirtualFile {
...
+    auto ret = new RegistryUploader(lfile, upload_url, username, password,
+                                    timeout, upload_bs, ctx);
     if (ret->init() < 0) {
         delete ret;
         return nullptr;
     }
     return ret;
 }
+
+photon::net::TLSContext *newTLSContextFromFile(const char *cert_file, const char *key_file) {
```
Comment (liulanzheng) [inline src/overlaybd/registryfs/registryfs_v2.cpp:811]: new_tls_context_from_file?

File: src/tools/overlaybd-apply.cpp [hunk 6]
```diff
@@ -44,6 +44,40 @@
...
+public:
+    IFile *m_fifo;
+
+    FIFOFile(IFile *fifo): m_fifo(fifo) { }
+    ~FIFOFile() { delete m_fifo; }
+
+    ssize_t read(void *buf, size_t count) override {
+        size_t left = count;
+        char *pos = (char *) buf;
+        LOG_INFO("read", VALUE(count));
```
Comment (liulanzheng) [inline src/tools/overlaybd-apply.cpp:57]: debug?

File: src/overlaybd/registryfs/registryfs_v2.cpp [hunk 7]
```diff
@@ -768,14 +787,65 @@ class RegistryUploader : public VirtualFile {
...
+    if (!cert_file || !key_file)
+        return photon::net::new_tls_context(nullptr, nullptr, nullptr);
+
+    if (strcmp(cert_file, "") == 0 || strcmp(key_file, "") == 0) {
+        return photon::net::new_tls_context(nullptr, nullptr, nullptr);
+    }
+
+    auto readFile = [](const char *fn, estring &target) -> int {
```
Comment (liulanzheng) [inline src/overlaybd/registryfs/registryfs_v2.cpp:819]: read_file

**Review by liulanzheng — APPROVED** [review]
LGTM

**Pull request merged**